src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -632,7 +663,7 @@ public static IResourceBuilder<T> WaitFor<T>(this IResourceBuilder<T> builder, I
 
             // If our dependency resource has health check annotations we want to wait until they turn healthy
             // otherwise we don't care about their health status.
-            if (dependency.Resource.TryGetAnnotationsOfType<HealthCheckAnnotation>(out var _))
+            if (includeHealthChecks && dependency.Resource.TryGetAnnotationsOfType<HealthCheckAnnotation>(out var _))
```

Comment:
This probably deserves a test that calls this with `includeHealthChecks: false` and makes sure it works correctly.

Reply:
added